presto-main/src/test/java/com/facebook/presto/execution/MockQueryExecution.java

```diff
@@ -44,26 +44,38 @@
     private final long memoryUsage;
     private final Duration cpuUsage;
     private final Session session;
+    private final QueryId queryId;
     private QueryState state = QUEUED;
     private Throwable failureCause;
 
     public MockQueryExecution(long memoryUsage)
     {
-        this(memoryUsage, 1);
+        this(memoryUsage, 1, "query_id");
     }
 
-    public MockQueryExecution(long memoryUsage, int priority)
+    public MockQueryExecution(long memoryUsage, String queryId)
     {
-        this(memoryUsage, new Duration(0, MILLISECONDS), priority);
+        this(memoryUsage, 1, queryId);
+    }
+
+    public MockQueryExecution(long memoryUsage, int priority, String queryId)
+    {
+        this(memoryUsage, new Duration(0, MILLISECONDS), priority, queryId);
     }
 
     public MockQueryExecution(long memoryUsage, Duration cpuUsage, int priority)
     {
+        this(memoryUsage, cpuUsage, priority, "query_id");
+    }
+
+    private MockQueryExecution(long memoryUsage, Duration cpuUsage, int priority, String queryId)
```

Comment:
When you have multiple overloads of constructors, that are effectively the same except some optional parameters, it gets messy pretty quickly, especially when the parameters are not aligned.

Originally, there was 3 versions:
```
MockQueryExecution(long memoryUsage)
MockQueryExecution(long memoryUsage, int priority)
MockQueryExecution(long memoryUsage, Duration cpuUsage, int priority)
```

Now you have:
```
1) MockQueryExecution(long memoryUsage)
2) MockQueryExecution(long memoryUsage, String queryId)                                         1
3) MockQueryExecution(long memoryUsage, int priority, String queryId)                           4
4) MockQueryExecution(long memoryUsage, Duration cpuUsage, int priority)                        2
5) MockQueryExecution(long memoryUsage, Duration cpuUsage, int priority, String queryId)
```

I propose having 3 overloads (note I reordered arguments):
```
MockQueryExecution(long memoryUsage)
MockQueryExecution(long memoryUsage, String queryId, int priority)
MockQueryExecution(long memoryUsage, String queryId, int priority, Duration cpuUsage)
```

We can only keep one of 3 and 4 to avoid messy overload. 2 is not really necessary given it has only 1 caller.